test/test_main.py

```diff
@@ -261,69 +262,63 @@ def test_event_not_implemented(self):
             self.fail('Event simulation for an unsupported event'
                       ' results in a non-zero exit code returned')
 
-    def test_setup_hooks(self):
+    def test_setup_event_links(self):
         """Test auto-creation of symlinks for supported events.
         """
-        event_hooks = [f'hooks/{e.replace("_", "-")}'
-                       for e in charm.Charm.on.events().keys()
-                       if e != 'install']
-
-        install_link_path = JUJU_CHARM_DIR / 'hooks/install'
-
-        # The symlink is expected to be present in the source tree.
-        self.assertTrue(install_link_path.exists())
-        # It has to point to main.py in the lib directory of the charm.
-        self.assertEqual(os.readlink(install_link_path), self.CHARM_PY_RELPATH)
-
-        def _assess_setup_hooks(event_spec):
-            event_hook = JUJU_CHARM_DIR / f'hooks/{event_spec.event_name}'
-
-            # Simulate a fork + exec of a hook from a unit agent.
-            self._simulate_event(event_spec)
-
-            r, _, files = next(os.walk(JUJU_CHARM_DIR / 'hooks'))
-
-            self.assertTrue(event_spec.event_name in files)
-
-            for event_hook in event_hooks:
-                self.assertTrue(os.path.exists(event_hook))
-                self.assertEqual(os.readlink(event_hook), 'install')
-                self.assertEqual(os.readlink('hooks/install'),
-                                 self.CHARM_PY_RELPATH)
+        event_hooks = [f'hooks/{e.replace("_", "-")}' for e in charm.Charm.on.events().keys()]
 
         charm_config = base64.b64encode(pickle.dumps({
             'STATE_FILE': self._state_file,
         }))
 
-        # Assess 'install' first because upgrade-charm or other
-        # events cannot be handled before install creates symlinks for them.
-        events_to_assess = (
-            EventSpec(InstallEvent, 'install', charm_config=charm_config),
-            EventSpec(StartEvent, 'start', charm_config=charm_config),
+        initial_combinations = {
+            'hooks_iaas': (
+                # 'install' comes first because upgrade-charm or other events cannot be handled before install creates symlinks for them.
+                EventSpec(InstallEvent, 'install', charm_config=charm_config),
+            ),
+            'hooks_iaas_storage': (
+                # Storage hooks run before "install" so this case needs to be checked as well.
```

Comment:
How is this case relevant for this test?

Reply:
Fixed in https://github.com/canonical/operator/pull/63/commits/bf4278d0cd3048231be3027cdbe9cd62264fa6b8